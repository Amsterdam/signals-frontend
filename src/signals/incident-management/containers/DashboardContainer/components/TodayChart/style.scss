.today-chart {
  align-self: center;
  text-align: center;

  .today-chart_date {
    font-size: 18px;
  }

  .today-chart_count {
    font-size: 180px;
  }
}
