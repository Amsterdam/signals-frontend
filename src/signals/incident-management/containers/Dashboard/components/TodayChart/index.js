import React from 'react';
import PropTypes from 'prop-types';
import moment from 'moment';


import './style.scss';

const TodayChart = ({ data }) => (
  <div className="today-chart">
    <div className="today-chart_date">{moment().format('dddd D MMMM YYYY')}</div>
    <div className="today-chart_count">{data.count}</div>
  </div>
);


TodayChart.defaultProps = {
  data: {}
};

TodayChart.propTypes = {
  data: PropTypes.object
};

export default TodayChart;
