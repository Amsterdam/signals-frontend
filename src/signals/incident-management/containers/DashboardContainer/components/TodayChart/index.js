import React from 'react';
import PropTypes from 'prop-types';
import moment from 'moment';


import './style.scss';

const TodayChart = ({ data }) => (
  <div className="today-chart">
    <div className="today-chart_date">{moment().format('dddd D MMMM YYYY')}</div>
    <div className="today-chart_count">{data}</div>
  </div>
);


TodayChart.defaultProps = {
  data: 0,
};

TodayChart.propTypes = {
  data: PropTypes.number,
};

export default TodayChart;
