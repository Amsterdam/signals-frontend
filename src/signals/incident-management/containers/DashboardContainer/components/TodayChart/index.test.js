import React from 'react';
import { shallow } from 'enzyme';
import moment from 'moment';

import TodayChart from './index';

jest.mock('moment');

describe('<TodayChart />', () => {
  let wrapper;
  let props;

  beforeEach(() => {
    moment.mockImplementation(() => ({
      format: () => 'vrijdag 11 januari 2019'
    }));

    props = {
      data: { count: 639 }
    };

    wrapper = shallow(
      <TodayChart {...props} />
    );
  });

  afterEach(() => {
    jest.resetAllMocks();
  });

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
