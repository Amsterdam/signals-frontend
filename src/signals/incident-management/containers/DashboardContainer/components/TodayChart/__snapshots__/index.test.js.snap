// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TodayChart /> should render correctly 1`] = `
<div
  className="today-chart"
>
  <div
    className="today-chart_date"
  >
    vrijdag 11 januari 2019
  </div>
  <div
    className="today-chart_count"
  >
    639
  </div>
</div>
`;
